src/archivematicaCommon/lib/elasticSearchFunctions.py

```diff
@@ -78,15 +75,83 @@
     'index': 'not_analyzed'
 }
 
+
 class ElasticsearchError(Exception):
+    """ Not operational errors. """
     pass
 
+
 class EmptySearchResultError(ElasticsearchError):
     pass
 
+
 class TooManyResultsError(ElasticsearchError):
     pass
 
+
+_es_hosts = None
+_es_client = None
+
+
+def setup(hosts):
+    """
+    Initialize Elasticsearch client and share it as the attribute _es_client in
+    the current module. An additional attribute _es_hosts is defined containing
+    the Elasticsearch hosts (expected types are: string, list or tuple).
+    """
+    global _es_hosts
+    global _es_client
+
+    _es_hosts = hosts
+    _es_client = Elasticsearch(**{
+        'hosts': _es_hosts,
+        'dead_timeout': 2
+    })
+
+
+def setup_reading_from_client_conf(config=None):
+    """
+    Similar to setup() but added to guarantee backward compatibility with old
+    consumers. Load the configuration from clientConfig.conf. It also accepts
+    an instance of ConfigParser, which is convenient when the consumer already
+    has loaded the configuration for other reasons.
+    """
+    if not config:
+        config = ConfigParser.SafeConfigParser()
+        config.read('/etc/archivematica/MCPClient/clientConfig.conf')
+    try:
+        hosts = config.get('MCPClient', "elasticsearchServer")
+    except (ConfigParser.NoSectionError, ConfigParser.NoOptionError):
+        hosts = '127.0.0.1:9200'
+    setup(hosts)
+
+
+def get_host():
+    """
+    Return one of the Elasticsearch hosts configured in our client. In the
+    future this function could look it up in the Elasticsearch client instead
+    of using the module attribute _es_hosts, because in an Elasticsearch
+    cluster, nodes can be added or removed dynamically.
+    """
+    if not _es_hosts:
+        raise ImproperlyConfigured('The Elasticsearch client has not been set up yet, please call setup() first.')
+    if isinstance(_es_hosts, (list, tuple)):
+        return _es_hosts[0]
+    return _es_hosts
+
+
+def get_client():
+    """
+    Obtain the current Elasticsearch client. If undefined, an exception will be
+    raised. This function also checks the integrity of the indices expected by
+    this application and populate them when they cannot be found.
+    """
+    if not _es_client:
+        raise ImproperlyConfigured('The Elasticsearch client has not been set up yet. Please call setup() first.')
```

Comment:
I think given the multiple ways to configure this, it probably doesn't make sense to automatically create the client, after thinking about it. Sorry for the noise!